ec2_scaling_group_template.json

```diff
@@ -0,0 +1,88 @@
+{
+  "Description":"Create an Amazon EC2 instances via scaling group running the Amazon Linux AMI.",
+  "Parameters":{
+    "KeyName":{
+      "Description":"Name of an existing EC2 KeyPair to enable SSH access to the instance",
+      "Type":"AWS::EC2::KeyPair::KeyName",
+      "Default":"default-admin",
+      "ConstraintDescription":"must be the name of an existing EC2 KeyPair."
+    },
+    "InstanceType":{
+      "Description":"WebServer EC2 instance type",
+      "Type":"String",
+      "Default":"t2.micro",
+      "AllowedValues":[
+        "t2.nano",
+        "t2.micro",
+        "t2.small"
+      ],
+      "ConstraintDescription":"must be a valid EC2 instance type."
+    }
+  },
+  "Resources":{
+    "mySecurityGroup": {
+      "Type": "AWS::EC2::SecurityGroup",
+      "Properties" : {
+        "GroupDescription" : "Allow http and ssh to client host",
+        "SecurityGroupIngress" : [
+          {
+            "IpProtocol" : "tcp",
+            "FromPort" : 80,
+            "ToPort" : 80,
+            "CidrIp" : "0.0.0.0/0"
+          },
+          {
+            "IpProtocol" : "tcp",
+            "FromPort" : 22,
+            "ToPort" : 22,
+            "CidrIp" : "0.0.0.0/0"
+          }
+        ]
+      }
+    },
+    "myLaunchTemplate": {
+      "Type": "AWS::EC2::LaunchTemplate",
+      "Properties": {
+        "LaunchTemplateName": { "Fn::Sub": "test-launch-template" },
+        "LaunchTemplateData": {
+          "ImageId": "ami-03ededff12e34e59e",
+          "InstanceType": { "Ref": "InstanceType" },
+          "KeyName": { "Ref": "KeyName" },
+          "SecurityGroupIds": [{ "Fn::GetAtt" : [ "mySecurityGroup", "GroupId" ] }],
+          "UserData": {
+            "Fn::Base64": {
+              "Fn::Join": [
+                "", [
+                "#!/bin/bash \n",
+                "yum update -y \n",
+                "yum -y install java-1.8.0-openjdk \n"
+              ]
+              ]
+            }
+          }
+        }
+      }
+    },
+    "myASG": {
+      "Type": "AWS::AutoScaling::AutoScalingGroup",
+      "Properties": {
+        "LaunchTemplate": {
+          "LaunchTemplateId": { "Ref": "myLaunchTemplate" },
+          "Version": { "Fn::GetAtt": [ "myLaunchTemplate", "LatestVersionNumber" ] }
+        },
+        "MaxSize": "2",
+        "MinSize": "2",
+        "DesiredCapacity": "2",
+        "AvailabilityZones": ["us-east-1a"]
+      }
+    }
+  },
+  "Outputs": {
```

Comment:
Just curious, why do you need SecurityGroupId as an output?

Reply:
As you know, whole this template is only for testing/learning. So I don't even need any resource from it)

Reply:
:) Outputs are usable even for testing/learning because they can provide i.e. Is of deployed resources. Instead of going to AWS Console UI, you can see right away the IP of your EC2 instances.